### PR TITLE
[processAutoBuy] Support bulkbuying

### PR DESCRIPTION
**Pull request by Mortimal**
Added checkItemBuyNeed to Misc.pm
Little changes to processAutoBuy for bulk buy support

NEED TESTING PLEASE!

### CODE REVIEW

**Comment by Mortimal** [comment]
Code is functioning... NEED TESTERS ih different game situations.

**Comment by alisonrag** [comment]
Can someone test this?

**Comment by fadreus** [comment]
image

Tested & not working.
Need updates?

**Comment by Mortimal** [comment]
@fadreus Is all ok without my code? I cant see dependency between error and my changes...

**Comment by fadreus** [comment]
@Mortimal It's fine without these files changes. Is it work on other server?

**Comment by alisonrag** [comment]
@fadreus 
this version of misc is old, you need to get a old version or copy only the subs for your openkore

**Comment by fadreus** [comment]
Tested those subs with CoreLogic & Misc from Oct 16th, `17; those error gone but when I ran autobuy, it talk to NPC, store window open up, done talked to NPC and then do nothing.

Tried with Fly, Butterfly & Awakening.

File: src/AI/CoreLogic.pm [hunk 1]
```diff
@@ -1904,51 +1896,55 @@ sub processAutoBuy {
...
+				  || (($i != $args->{lastIndex}) && !checkItemBuyNeed($i,$zenyleft)));
+			$item = $storeList->getByName( $config{"buyAuto_".$i} );
+			if (defined $item){
+				$args->{'nameID'} = $item->{nameID};
+			}
+			if (!exists $args->{'nameID'}){
+				$args->{index_failed}{$i} = 1;
+				error "buyAuto index ".$i." (".$config{"buyAuto_".$i}.") failed, item doesn't exist in npc sell list.\n", "npc";
+			}else{
+				my $maxbuy = ($config{"buyAuto_".$i."_price"}) ? int($zenyleft/$config{"buyAuto_".$i."_price"}) : 30000; # we assume we can buy 30000, when price of the item is set to 0 or undef
```
Comment (Nipodemos) [inline src/AI/CoreLogic.pm]: i think it is a little danger to set the amout to be the max that money can buy
so if a new user forgot do put maxAmaount, and it is a char with a LOT of zeny
openkore will try to buy 30000 of the item? very dangerous

Comment (Nipodemos) [inline src/AI/CoreLogic.pm]: i think it is a little danger to set the amout to be the max that money can buy
so if a new user forgot do put maxAmaount, and it is a char with a LOT of zeny
openkore will try to buy 30000 of the item? very dangerous

**Comment by Fadreus** [comment]
@Mortimal Will you check this again?

**Comment by Nipodemos** [comment]
for now i'll close this pull request, since @Mortimal has quit openkore
but i'm not deleting , so it will be recoverable

File: src/AI/CoreLogic.pm (hunk 1 above)
Comment (Mortimal) [reply src/AI/CoreLogic.pm]: But.... But.... But.... it was in original logic ( k il change that

**Comment by Mortimal** [comment]
DELETE THIS i will do one more

**Pull request closed without merging**